### PR TITLE
Implement Frame Saliency Attack

### LINKED ISSUE #353: Implement adversarial attacks for optical flow-based action recognition classifiers
Implement adversarial attacks for optical flow-based action recognition classifiers

Paper link: https://arxiv.org/abs/1811.11875

### PR DESCRIPTION
**Pull request by mathsinn**
# Description

Implemented the Frame Saliency Attack in a new evasion attack module (issue #353). To support it, implemented masks that can be used in FGM and PGD generation of adversarial examples. Also added new util function for computing attack success boolean array for given adversarial examples and attack objectives. Finally, added demonstration of the new attack to the Action Recognition Attack notebook.

Fixes #353

## Type of change

Please check all relevant options.

- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] Implemented and ran unit tests for the new / modified capabilities
- [x] Implemented and ran end-to-end test in sample notebook

**Test Configuration**:
- OS: MacOS
- Python version 3.7.4
- ART version or commit number 1.2.0
- TensorFlow 1.13.1 / Keras 2.2.4 / MXNet 1.6.0

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by mathsinn** [comment]
Thank you @beat-buesser for the suggestion. I have made those additions.
